Sources/Application/MainCoordinator.swift

```diff
@@ -42,19 +43,23 @@ final class MainCoordinator: NSObject,
     }
     
     func evaluateRevisit() {
-        if userStore.validAuthenticatedUser {
-            Task(priority: .userInitiated) {
-                await MainActor.run {
-                    do {
-                        let idToken = try userStore.readItem(itemName: .idToken,
-                                                             storage: .authenticated)
-                        tokenHolder.idTokenPayload = try tokenVerifier.extractPayload(idToken)
-                        updateToken()
-                        windowManager.hideUnlockWindow()
-                    } catch {
-                        handleLoginError(error)
+        if userStore.previouslyAuthenticatedUser != nil {
+            if userStore.validAuthenticatedUser {
+                Task(priority: .userInitiated) {
+                    await MainActor.run {
+                        do {
+                            let idToken = try userStore.readItem(itemName: .idToken,
+                                                                 storage: .authenticated)
+                            tokenHolder.idTokenPayload = try tokenVerifier.extractPayload(idToken)
+                            updateToken()
+                            windowManager.hideUnlockWindow()
+                        } catch {
+                            handleLoginError(error)
+                        }
                     }
                 }
+            } else {
+                fullLogin(error: TokenError.expired)
```

Comment:
This flow is now driven from two points in the `userStore`:
1. If the user has any accessTokenExpiry date stored in `UserDefaults` at all - telling us whether we have any record of them previously logging in before.
2. If the user has an expired accessTokenExpiry date stored in `UserDefaults` - telling us that they have logged in before but their token is now expired and they need to do a full log in to use the app.

This allows us to display a sign out warning screen over the intro screen if we can see that they have previously authenticated, but their access token has now expired. Rather than someone who has not authenticated before and thus no sign out warning screen is needed.